docs/content/intro/tutorial-preview-environments.md

```diff
@@ -0,0 +1,424 @@
+  await createNamespace(environmentName);
+};
+
+events.on("exec", event => {
+    const payload = JSON.parse(event.payload);
+    const { name } = payload;
+
+    if (!name) {
+      throw Error("Environment name must be specified");
+    }
+    if (protectedEnvironment(name)) {
+      throw Error(`Environment '${name}' is protected`);
+    }
+    provisionEnvironment(name, projects).catch(error => {
+      throw error;
+    });
+});
+
+```
+
+In the same directory add `payload.json` file with this content:
+
+```json
+{"name": "bob"}
+```
+
+Let's run our workflow with `brig`:
+
+```console
+$ brig run brigadecore/brigade-tutorial-config -f brigade.js -p payload.json --namespace brigade
+```
+
+Command above will trigger brigade workflow by executing `brigade.js` script with a data from `payload.json`.
+
+At this point our `brigade.js` script will simply use Kubernetes API to create new namespace. Name of new namespace is set in payload json.
+
+We first comparing it with a list of `protected` namespace names and then we're using 
+API to check if the namespace already exists, and if not, we create it. 
+
+> **With power of Kubernetes API we can easily orchestrate any aspects of our pipeline.**
+
+## Environment Dependencies
+
+In this step we'll automate adding external dependencies like PostgreSQL to our new environment.
+
+```js
+const k8sAppClient = kubeConfig.makeApiClient(kubernetes.Apps_v1Api);
+
+const deployDependencies = async environmentName => {
+  const postgresqlStatefulSet = await k8sAppClient.listNamespacedStatefulSet(
+    environmentName,
+    undefined,
+    undefined,
+    undefined,
+    undefined,
+    "app=postgresql",
+  );
+  if (postgresqlStatefulSet.body.items.length) {
+    console.log("postgresql already deployed");
+  } else {
+    const postgresql = new Job("postgresql", "lachlanevenson/k8s-helm:v2.12.3");
+    postgresql.storage.enabled = false;
+    postgresql.imageForcePull = true;
+    postgresql.tasks = [
+      `helm init --client-only && \
+      helm repo update && \
+      helm upgrade ${environmentName}-postgresql stable/postgresql \
+      --install --namespace=${environmentName} \
+      --set fullnameOverride=postgresql \
+      --set postgresqlDatabase=products \
+      --set resources.requests.cpu=50m \
+      --set resources.requests.memory=156Mi \
+      --set readinessProbe.initialDelaySeconds=60 \
+      --set livenessProbe.initialDelaySeconds=60;`,
+    ];
+    await postgresql.run();
+  }
+};
+
+const provisionEnvironment = async (environmentName, projects) => {
+  await deployDependencies(environmentName);
+};
+
+```
+
+Before deploying our PostgreSQL StatefulSet we check if one already exists. 
+To list StatefulSets in a Namespace we need to use `listNamespacedStatefulSet` 
+api endpoint that lives in `kubernetes.Apps_v1Api` set of APIs. 
+We create instance of it called `k8sAppClient` first.
+
+Once we verify our dependency needs to be installed we create `postgresql` Job that will 
```

Comment:
Grammatical/punctuation amendments: `...needs to be installed, we create a postgresql Job that will run the lachlanevenson/k8s-helm:v2.12.3 image as its worker`